File: Project-Euler-Java/ProjectEuler004.java
public class ProjectEuler004 {
    public String run() {
        int answer = 0;
        for (int i = 999; i >= 100; i--) {
            for (int j = 999; j >= 100; j--) {
                int product = i * j;
                if (Library.isPalindrome(product) && product > answer) answer = product;
            }
        }
        return String.valueOf(answer);
    }

    public static void main(String[] args) {
        System.out.println(new ProjectEuler004().run());
    }




}

File: Project-Euler-Python/ProjectEuler004.py
# Check if n is palindrome
def is_palindrome(n):
    n = str(n)
    reverse_n = n[::-1]
    if n == reverse_n:
        return True
    return False


answer = 0
for i in range(999, 99, -1):
    for j in range(999, 99, -1):
        product = i * j
        if is_palindrome(product) and product > answer:
            answer = product

print(answer)
